lib/src/database/db.dart

```diff
@@ -209,7 +209,12 @@ class Db {
       }
 
       if (connection == null) {
-        connection = _masterConnection;
+        if (_masterConnection != null) {
+          connection = _masterConnection;
+        }
+        else {
+          throw new MongoDartError('No master connection');
```

Comment:
Can you extract this message into a constant. And also, you repeat this exact piece of code 3 times. You could extract a method from that and call that instead.